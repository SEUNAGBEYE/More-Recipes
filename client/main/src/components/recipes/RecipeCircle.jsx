import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
/**
 * @description - Recipe Circle Component
 *
 * @param {Object} props
 *
 * @returns {Jsx} Jsx
 */
const RecipeCircle = (props) => (
  <div className="col-sm-6 col-md-4"
    style={{ fontWeight: 'bold', textAlign: 'center' }}
  >
    <Link to={props.to || `recipe/${props.recipe.id}`}>
      <div className="box">
        <div className="circle">
          <img className ="circle" src= {props.recipe.image} />
        </div>
      </div>
      <div className="break-word">{ props.recipe.name }</div>
    </Link>
  </div>
);

const propTypes = {
  to: PropTypes.string,
  recipe: PropTypes.object.isRequired
};

RecipeCircle.propTypes = propTypes;

export default RecipeCircle;
