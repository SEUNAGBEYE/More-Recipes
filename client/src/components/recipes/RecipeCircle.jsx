import React from 'react';
import { Link } from 'react-router-dom';

const RecipeCircle = (props) => (
  <div className="col-sm-6 col-md-4" style={{ fontWeight: 'bold', textAlign: 'center' }}>
    <Link to={props.to || `recipe/${props.recipe.id}`}>
      <div className="box">
        <div className="circle"><img className ="circle" src= {props.recipe.image} /></div>
      </div>
      <div>{ props.recipe.name }</div>
    </Link>
  </div>
);

export default RecipeCircle;
