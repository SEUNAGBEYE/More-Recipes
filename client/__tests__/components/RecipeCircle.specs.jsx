

import React from 'react';
import { shallow } from 'enzyme';
import RecipeCircle from '../../main/src/components/recipes/RecipeCircle';

const recipe = {
  userId: 1,
  name: 'Amala',
  image: 'This is the image',
  upvotes: [1,2,4],
  downvotes: [1,2,4],
  description: 'This is the decription',
  views: [1,2]
}

const props = {
  recipe
};
  

describe('# RecipeCircle', () => {
  const wrapper = shallow(
      <RecipeCircle {...props} />
  );
  it('should render successfully', (done) => {
    expect(wrapper).toBeDefined();
    expect(wrapper.getElement().type).toBe('div');
    expect(wrapper).toMatchSnapshot();
    done();
  });
});
