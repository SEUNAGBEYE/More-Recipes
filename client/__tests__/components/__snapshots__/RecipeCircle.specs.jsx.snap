// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# RecipeCircle should render successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecipeCircle
    recipe={
      Object {
        "description": "This is the decription",
        "downvotes": Array [
          1,
          2,
          4,
        ],
        "image": "This is the image",
        "name": "Amala",
        "upvotes": Array [
          1,
          2,
          4,
        ],
        "userId": 1,
        "views": Array [
          1,
          2,
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        replace={false}
        to="recipe/undefined"
      >
        <div
          className="box"
        >
          <div
            className="circle"
          >
            <img
              className="circle"
              src="This is the image"
            />
          </div>
        </div>
        <div
          className="break-word"
        >
          Amala
        </div>
      </Link>,
      "className": "col-sm-6 col-md-4",
      "style": Object {
        "fontWeight": "bold",
        "textAlign": "center",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="box"
          >
            <div
              className="circle"
            >
              <img
                className="circle"
                src="This is the image"
              />
            </div>
          </div>,
          <div
            className="break-word"
          >
            Amala
          </div>,
        ],
        "replace": false,
        "to": "recipe/undefined",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="circle"
            >
              <img
                className="circle"
                src="This is the image"
              />
            </div>,
            "className": "box",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                className="circle"
                src="This is the image"
              />,
              "className": "circle",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "circle",
                "src": "This is the image",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Amala",
            "className": "break-word",
          },
          "ref": null,
          "rendered": "Amala",
          "type": "div",
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          replace={false}
          to="recipe/undefined"
        >
          <div
            className="box"
          >
            <div
              className="circle"
            >
              <img
                className="circle"
                src="This is the image"
              />
            </div>
          </div>
          <div
            className="break-word"
          >
            Amala
          </div>
        </Link>,
        "className": "col-sm-6 col-md-4",
        "style": Object {
          "fontWeight": "bold",
          "textAlign": "center",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="box"
            >
              <div
                className="circle"
              >
                <img
                  className="circle"
                  src="This is the image"
                />
              </div>
            </div>,
            <div
              className="break-word"
            >
              Amala
            </div>,
          ],
          "replace": false,
          "to": "recipe/undefined",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="circle"
              >
                <img
                  className="circle"
                  src="This is the image"
                />
              </div>,
              "className": "box",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  className="circle"
                  src="This is the image"
                />,
                "className": "circle",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "circle",
                  "src": "This is the image",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Amala",
              "className": "break-word",
            },
            "ref": null,
            "rendered": "Amala",
            "type": "div",
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
